File: api/internal/domain/models/league_member.go
package models

import "time"

type LeagueMember struct {
	LeagueID   string    `json:"leagueId"`
	PlayerID   string    `json:"playerId"`
	PlayerName string    `json:"playerName"`
	Role       Role      `json:"role"`
	JoinedAt   time.Time `json:"joinedAt"`
}

func CreateLeagueMember(leagueID, playerID string, role Role) *LeagueMember {
	return &LeagueMember{
		LeagueID: leagueID,
		PlayerID: playerID,
		Role:     role,
		JoinedAt: time.Now(),
	}
}

File: bouncy-ui/lib/models/league_member.dart
class LeagueMember {
  final String leagueId;
  final String playerId;
  final String role;
  final DateTime joinedAt;

  LeagueMember({
    required this.leagueId,
    required this.playerId,
    required this.role,
    required this.joinedAt,
  });

  factory LeagueMember.fromMap(Map<String, dynamic> map) {
    return LeagueMember(
      leagueId: map['LeagueID'] ?? '',
      playerId: map['PlayerID'] ?? '',
      role: map['Role']?.toString() ?? '',
      joinedAt: DateTime.tryParse(map['JoinedAt'] ?? '') ?? DateTime.now(),
    );
  }
}